lib/model/store.dart

```diff
@@ -778,19 +778,22 @@ class LiveGlobalStore extends GlobalStore {
 
   /// The file path to use for the app database.
   static Future<File> _dbFile() async {
-    // What directory should we use?
-    //   path_provider's getApplicationSupportDirectory:
-    //     on Android, -> Flutter's PathUtils.getFilesDir -> https://developer.android.com/reference/android/content/Context#getFilesDir()
-    //       -> empirically /data/data/com.zulip.flutter/files/
-    //     on iOS, -> "Library/Application Support" via https://developer.apple.com/documentation/foundation/nssearchpathdirectory/nsapplicationsupportdirectory
-    //     on Linux, -> "${XDG_DATA_HOME:-~/.local/share}/com.zulip.flutter/"
-    //     All seem reasonable.
-    //   path_provider's getApplicationDocumentsDirectory:
-    //     on Android, -> Flutter's PathUtils.getDataDirectory -> https://developer.android.com/reference/android/content/Context#getDir(java.lang.String,%20int)
-    //       with https://developer.android.com/reference/android/content/Context#MODE_PRIVATE
-    //     on iOS, "Document directory" via https://developer.apple.com/documentation/foundation/nssearchpathdirectory/nsdocumentdirectory
-    //     on Linux, -> `xdg-user-dir DOCUMENTS` -> e.g. ~/Documents
-    //     That Linux answer is definitely not a fit.  Harder to tell about the rest.
+    // path_provider's getApplicationSupportDirectory:
+    //   on Android, -> Flutter's PathUtils.getFilesDir -> https://developer.android.com/reference/android/content/Context#getFilesDir()
+    //     -> empirically /data/data/com.zulip.flutter/files/
+    //   on iOS, -> "Library/Application Support" via https://developer.apple.com/documentation/foundation/nssearchpathdirectory/nsapplicationsupportdirectory
+    //   on Linux, -> "${XDG_DATA_HOME:-~/.local/share}/com.zulip.flutter/"
+    //
+    // This is reasonable for iOS per Apple's recommendation:
```

Comment:
How about "This is the best fit for iOS per Apple's recommendation", if that's true? That would feel more satisfying than just saying it's a reasonable option, because then you wonder if there are other reasonable options and what those look like.

Same for Android and Linux below, and I don't yet follow your reasoning about Android. (Linux is clear from the removed line "That Linux answer is definitely not a fit.")

Reply:
Will put the relevant pieces back here. I dug into the path for Android more, and found that there is [`getDatabasePath`](https://developer.android.com/reference/android/content/Context#getDatabasePath(java.lang.String)), which is not provided by path_provider. It would be nice to have support for that, though.

Reply:
Opened #1321 and refer to the issue instead. We should be good with using `getApplicationSupportDirectory` for this PR.

Reply:
Will be skipping this commit after [a CZO discussion](https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/.23F1248.20database.20path.20for.20Android).